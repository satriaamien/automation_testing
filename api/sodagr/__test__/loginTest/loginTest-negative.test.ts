import dotenv from "dotenv";
dotenv.config();
const username = process.env.usernameSodagr;
const password = process.env.passwordSodagr;
import {
  login,
  loginWrongTypeEmail,
  loginWrongTypePassword,
} from "../../endpoint/login_test";
describe("user login with invalid credential", () => {
  test("login with valid email & invalid type password", async () => {
    const payload = {
      email: 982345678,
      password: `${password}`,
    };
    const response = await loginWrongTypeEmail(payload);
    expect(response.body).toMatchSnapshot();
  });
  test("login with invalid type email & valid password", async () => {
    const payload = {
      email: `${username}`,
      password: 9123456789,
    };
    const response = await loginWrongTypePassword(payload);
    expect(response.body).toMatchSnapshot();
  });
  test("login with invalid type email & password", async () => {
    const payload = {
      email: "",
      password: "",
    };
    const response = await login(payload);
    expect(response.body).toMatchSnapshot();
  });
});
