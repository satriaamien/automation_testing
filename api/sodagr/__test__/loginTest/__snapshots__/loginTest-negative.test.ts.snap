// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user login with invalid credential login with invalid type email & password 1`] = `
{
  "message": "email: email must be an email",
  "success": false,
}
`;

exports[`user login with invalid credential login with invalid type email & valid password 1`] = `
{
  "message": "password: password must be a string",
  "success": false,
}
`;

exports[`user login with invalid credential login with valid email & invalid type password 1`] = `
{
  "message": "email: email must be an email",
  "success": false,
}
`;
